pages/index.tsx

```diff
@@ -11,10 +11,10 @@ import Error from "./_error";
 
 interface IProps {
     candidate: { id: string; username: string } | undefined;
-    role: string;
-    token: string;
-    user: { id: string; username: string };
-    user_id: number;
+    role: string | undefined;
```

Comment:
No, if you don't have required props, you should throw an error in getInitialProps